core/src/com/fourinachamber/fortyfive/screen/general/CustomActors.kt

```diff
@@ -533,6 +532,282 @@ open class CustomFlexBox(
+            val parentPos = this@CustomScrollableFlexBox.localToStageCoordinates(Vector2(0, 0))
+            startPos = if (isScrollDirectionVertical) {
+                val relX = (if (scrollbarSide == "left") 0F else width - scrollbarHandle.width)
+                val relY = scrollbarHandle.y - parentPos.y
+                if (touchDownX < relX + scrollbarHandle.width && touchDownX > relX
+                    && touchDownY < relY + scrollbarHandle.height && touchDownY > relY
+                ) scrollbarHandle.y
+                else Float.NaN
+            } else {
+                val relX = scrollbarHandle.x - parentPos.x
+                val relY = (if (scrollbarSide == "top") height - scrollbarHandle.height else 0F)
+                if (touchDownX < relX + scrollbarHandle.width && touchDownX > relX
+                    && touchDownY < relY + scrollbarHandle.height && touchDownY > relY
+                ) scrollbarHandle.x
+                else Float.NaN
+            }
+        }
+
+        override fun drag(event: InputEvent?, x: Float, y: Float, pointer: Int) {
+            if (startPos.isNaN()) return
+            val parentPos = this@CustomScrollableFlexBox.localToStageCoordinates(Vector2(0, 0))
+            if (isScrollDirectionVertical) {
+                val max = lastMax - cutBottom
+                val curSize = height * height / (max + height)
+                scrollbarHandle.y = startPos + (y - touchDownY) + cutTop
+                offset = (-(scrollbarHandle.y - parentPos.x) - curSize) * max / (height - curSize)
+            } else {
+                scrollbarHandle.x = (startPos + x - touchDownX)
+                val max = lastMax
+                val curSize = width * width / (max + width)
+                offset = -(scrollbarHandle.x - parentPos.x) * max / (width - curSize) - cutLeft
+            }
+            invalidate()
+        }
+    }
+
+    init {
+        addListener(scrollListener)
+        addListener(dragListener)
+    }
+
+    private fun scroll(offset: Float) {
+        this.offset += offset * scrollDistance
+        invalidate()
+    }
+
+    private var offset: Float = 0F
+    var cutLeft: Float = 0F
+    var cutRight: Float = 0F
+    var cutTop: Float = 0F
+    var cutBottom: Float = 0F
+    private var lastMax: Float = 0F
+    private lateinit var scrollbarBackground: CustomImageActor
+    private lateinit var scrollbarHandle: CustomImageActor
+    var scrollbarWidth: Float = 3F
+
+    override fun layout() {
+        super.layout()
+
+        layoutChildren()
+        layoutScrollBar()
+    }
+
+    private fun layoutScrollBar() {
+        if (scrollbarBackgroundName != null) {
+            if (!this::scrollbarBackground.isInitialized) {
+                scrollbarBackground = screen.namedActorOrError(scrollbarBackgroundName) as CustomImageActor
+                remove(scrollbarBackground.styleManager?.node)
+            }
+            if (needsScrollbar)
+                layoutScrollbarBackground()
```

Comment:
Not a big deal, but when I prefer if-statement to have its body on the same line when no braces are used